proxycore/conn.go

```diff
@@ -95,6 +96,7 @@ func NewConn(conn net.Conn, recv Receiver) *Conn {
 		conn:     conn,
 		recv:     recv,
 		writer:   bufio.NewWriterSize(conn, MaxCoalesceSize),
+		reader:   bufio.NewReader(conn),
```

Comment:
I see no meaningful difference using `bufio.NewReaderSize(conn, MaxCoalesceSize)`:

```
num_requests,   duration, final rate,       min,       mean,     median,       75th,       95th,       98th,       99th,     99.9th,        max
    10000000,     40.658,     245954,       517,      20321,      16479,      27423,      46463,      55839,      62463,      91519,     147583

num_requests,   duration, final rate,       min,       mean,     median,       75th,       95th,       98th,       99th,     99.9th,        max
    10000000,    40.3951,     247554,       458,      20190,      17231,      26463,      44255,      53503,      60543,      82751,     142335

num_requests,   duration, final rate,       min,       mean,     median,       75th,       95th,       98th,       99th,     99.9th,        max
    10000000,    41.1132,     243231,       462,      20545,      17263,      27343,      46591,      55295,      61727,      84607,     152703
```

```diff
diff --git a/proxycore/conn.go b/proxycore/conn.go
index af807fd..f686842 100644
--- a/proxycore/conn.go
+++ b/proxycore/conn.go
@@ -96,7 +96,7 @@ func NewConn(conn net.Conn, recv Receiver) *Conn {
                conn:     conn,
                recv:     recv,
                writer:   bufio.NewWriterSize(conn, MaxCoalesceSize),
-               reader:   bufio.NewReader(conn),
+               reader:   bufio.NewReaderSize(conn, MaxCoalesceSize),
```

Reply:
Makes me want to reevaluate using `bufio.NewWriterSize(conn, MaxCoalesceSize)`.